### PR TITLE
PPF-593 - Add guard for consistency integration type

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Added

- [Add guard for consistency integration type](https://github.com/cultuurnet/publiq-platform/commit/f7fceb90e5f3c5fb546321d77fc5ddb8505a2d26)

### Fixed

- Fix failing tests

---
Ticket: https://jira.uitdatabank.be/browse/PPF-593

### CODE REVIEW

File: app/Domain/Integrations/Exceptions/InconsistentIntegrationTypeException.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+declare(strict_types=1);
+
+namespace App\Domain\Integrations\Exceptions;
+
+use App\Domain\Integrations\Integration;
+use App\Domain\Subscriptions\Subscription;
+use RuntimeException;
+
+final class InconsistentIntegrationTypeException extends RuntimeException
```
Comment (LucWollants) [inline app/Domain/Integrations/Exceptions/InconsistentIntegrationTypeException.php:11]: No need to add the `Exception` suffix. Same reason as why not to add `Interface`, `Abstract`, etc

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php [hunk 2]
```diff
@@ -192,4 +196,23 @@ private function saveTransaction(Integration $integration, ?string $couponCode):
             }
         });
     }
+
+    /**
+     * @throws InconsistentIntegrationTypeException
+     */
+    private function guardConsistencyIntegrationType(UuidInterface $subscriptionId, Integration $integration): void
```
Comment (LucWollants) [inline app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php:203]:
```suggestion
    private function guardIntegrationTypeConsistency(UuidInterface $subscriptionId, Integration $integration): void
```

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php [hunk 3]
```diff
@@ -192,4 +196,23 @@ private function saveTransaction(Integration $integration, ?string $couponCode):
...
+    {
+        /** @var SubscriptionModel $subscriptionModel */
+        $subscriptionModel = SubscriptionModel::query()->findOrFail($subscriptionId->toString());
```
Comment (LucWollants) [inline app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php:206]: We already inject the `EloquentUdbOrganizerRepository` so I think it's fine to inject the 
`SubscriptionRepository` https://github.com/cultuurnet/publiq-platform/pull/1362/files#diff-d7f8c09ae0d8e31ca6a38cb4d3b676372a2dbaefae83d66772e86285d88a6532L21

File: app/Domain/Integrations/Exceptions/InconsistentIntegrationTypeException.php (hunk 1 above)
Comment (simon-debruijn) [reply app/Domain/Integrations/Exceptions/InconsistentIntegrationTypeException.php]: [Remove Exception suffix](https://github.com/cultuurnet/publiq-platform/pull/1362/commits/631ae15bdc135e20c5b986656375d6727f7c75ca)

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php (hunk 2 above)
Comment (simon-debruijn) [reply app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php]: [Rename guardIntegrationTypeConsistency](https://github.com/cultuurnet/publiq-platform/pull/1362/commits/fce35ed08487b6351c15734f4851c32b5a93beaf)

**Review by LucWollants — APPROVED** [review]
Thanks for the extra tweaks.

**Pull request merged**